src/lightning/fabric/fabric.py

```diff
@@ -574,19 +574,30 @@ def no_backward_sync(self, module: _FabricModule, enabled: bool = True) -> Gener
     def sharded_model(self) -> Generator:
         """Shard the parameters of the model instantly when instantiating the layers.
 
-        Use this context manager with strategies that support sharding the model parameters to save peak memory usage.
+        .. deprecated:: This context manager is deprecated in favor of :meth:`init_module`, use it instead.
+        """
+        rank_zero_deprecation("`Fabric.sharded_model()` is deprecated in favor of `Fabric.init_module()`.")
```

Comment:
This should be mentioned in the CHANGELOG, deprecated section.

Since CI is green and this is ready to go, feel free to add this in a different PR :)

Reply:
thanks, added in #17287